src/crimson/osd/recovery_backend.h

```diff
@@ -141,11 +145,14 @@ class RecoveryBackend {
     public boost::intrusive_ref_counter<
       WaitForObjectRecovery, boost::thread_unsafe_counter>,
     public crimson::BlockerT<WaitForObjectRecovery> {
+      crimson::osd::PG &pg;
     std::optional<seastar::shared_promise<>> readable, recovered, pulled;
     std::map<pg_shard_t, seastar::shared_promise<>> pushes;
   public:
     static constexpr const char* type_name = "WaitForObjectRecovery";
 
+    WaitForObjectRecovery(crimson::osd::PG &pg) : pg(pg) {}
```

Comment:
Is `pg` passed to guarantee lifetime?

Reply:
Yes, `pg` is guaranteed to outlive `WaitForObjectRecovery`.